src/core/namespace.c

```diff
@@ -1305,6 +1359,79 @@ int bind_mount_add(BindMount **b, unsigned *n, const BindMount *item) {
         return 0;
 }
 
+void temporary_filesystem_free_many(TemporaryFileSystem *t, unsigned n) {
+        unsigned i;
+
+        assert(t || n == 0);
+
+        for (i = 0; i < n; i++) {
+                free(t[i].path);
+                free(t[i].options);
+        }
+
+        free(t);
+}
+
+int temporary_filesystem_add(
+                TemporaryFileSystem **t,
+                unsigned *n,
+                bool ignore_enoent,
+                const char *path,
+                const char *options) {
+
+        _cleanup_free_ char *p = NULL, *o = NULL, *flag_string = NULL;
+        unsigned long flags = MS_NODEV|MS_STRICTATIME;
+        TemporaryFileSystem *c;
+        bool read_only;
+        char *a;
+        int r;
+
+        assert(t);
+        assert(n);
+        assert(path);
+
+        p = strdup(path);
+        if (!p)
+                return -ENOMEM;
+
+        r = mnt_optstr_get_flags(options, &flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+        if (r < 0)
+                return r;
+
+        r = mnt_optstr_apply_flags(&flag_string, flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+        if (r < 0)
+                return r;
```

Comment:
I think we should move the mount option processing to some call in moun-util.[ch] or so, after all i figure nspawn should use that too in the long run... maybe some function: `mount_option_mangle(const char *options, unsigned long *ret_flag_options, char **ret_remaining_options)` that hides the mangling away, and removes all bit flag options from the string in one go

Reply:
Yeah, nspawn-mount.c needs to be updated. I thought I will do in another PR. But, now, yes, I will do that in this PR.

Reply:
Updated.